caikit/core/toolkit/destroyable_process.py

```diff
@@ -0,0 +1,193 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A DestroyableProcess implements a multiprocessing Process that captures errors
+and communicates them to its parent with clean semantics for being destroyed.
+
+NOTE: The "fork" start method is used for two reasons:
+    1. It's faster
+    2. For auto-generated classes (e.g. stream sources), "spawn" can result in
+        missing classes since it performs a full re-import, but may not
+        regenerate these classes.
+"""
+# Standard
+from functools import partial
+from typing import Any, Callable, Optional, Tuple
+import multiprocessing
+
+# First Party
+import alog
+
+# Local
+from .destroyable import Destroyable
+
+log = alog.use_channel("DESTROY-PROC")
+
+FORK_CTX = multiprocessing.get_context("fork")
+
+
```

Comment:
It seems we have removed the patch required for python forks to work correctly inside a Threadpoolexecutor. Ref: https://github.com/caikit/caikit/blob/main/caikit/runtime/work_management/train_executors.py#L50

Reply:
This is actually included in the `LocalTrainer` (next PR): https://github.com/caikit/caikit/pull/310/files#diff-01866f980d3c84ae181021e96cf0ccd560747994902c657fb5fad90e3edde31fR52